Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Use the official Golang image to build the Go binary
+FROM golang:1.22 AS builder
+
+# Set the Current Working Directory inside the container
+WORKDIR /app
+
+# Copy go.mod and go.sum files and download dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy the source code into the container
+COPY . .
+
+# Build the Go app
+RUN GOOS=linux GOARCH=amd64 go build -o main cmd/productivite/main.go
+
+# Use a minimal base image to run the Go binary
+FROM alpine:latest
+
+# Install necessary packages
+RUN apk --no-cache add ca-certificates
+
+# Set the Current Working Directory inside the container
+WORKDIR /root/
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider using non-root user for security</b></div>
<div id="fix">

Consider using a non-root user for running the application. Running as root is generally considered a security risk.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -20,14 +20,16 @@
 RUN apk --no-cache add ca-certificates

 # Create non-root user
 +RUN adduser -D appuser
 +
 # Set the Current Working Directory inside the container
 -WORKDIR /root/
 +WORKDIR /home/appuser

 # Copy the pre-built binary from the builder stage
 COPY --from=builder /app/main .

 # Ensure the binary is executable and set ownership
 -RUN chmod +x /root/main
 +RUN chown appuser:appuser main && chmod +x main
 +
 +USER appuser
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/productivite/pull/13#issuecomment-2556339094>#b3212a</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged